pkg/kubernetes/reconcile.go

```diff
@@ -14,138 +14,129 @@ import (
 	"github.com/grafana/tanka/pkg/spec/v1alpha1"
 )
 
-// Reconcile extracts all valid Kubernetes objects from the raw output of the
-// Jsonnet compiler. A valid object is identified by the presence of `kind` and
-// `apiVersion`.
-// TODO: Check on `metadata.name` as well and assert that they are
-// not only set but also strings.
-func Reconcile(raw map[string]interface{}, spec v1alpha1.Spec, targets []*regexp.Regexp) (state manifest.List, err error) {
-	extracted, err := extract(raw)
+// Manifest is a type alias so we can use local variable name manifest without colliding
+// with the package
+type Manifest = manifest.Manifest
+
+// Reconcile extracts kubernetes Manifests from raw evaluated jsonnet <kind>/<name>,
+// provided the manifests match the given regular expressions. It finds each manifest by
+// recursively walking the jsonnet structure.
+//
+// In addition, we sort the manifests to ensure the order is consistent in each
+// show/diff/apply cycle. This isn't necessary, but it does help users by producing
+// consistent diffs.
+func Reconcile(raw map[string]interface{}, spec v1alpha1.Spec, kindNameMatchers []*regexp.Regexp) (state manifest.List, err error) {
```

Comment:
What's wrong with calling them `objectspecs` (**object** **spec**ifications), or targets which is what they are called at the command line? (you specify your target set using objectspecs)

Reply:
What is an object spec? It's definitely not a string of `<kind>/<name>`, it's a specification file that is uniquely identified by its group-version-kind-namespace-name, but that's not what we're using here.

I think objectspec might be something you've come up with internally at Grafana? I renamed this to be kindName because then anyone, even engineers outside of Grafana, could see immediately what this variable may be.

Reply:
Hrm, `objectspec` was just something I came up with during designing this feature. Was probably inspired by the term `pathspec` used by `git` to describe everything that notes the location of a ref.

Happy to leave it as `kindNameMatchers`!